modules/loader/src/main/java/com/opengamma/strata/loader/csv/FullSwapTradeCsvPlugin.java

```diff
@@ -266,6 +277,39 @@ private static RateCalculationSwapLeg parseLeg(
     PeriodicSchedule accrualSch = parseAccrualSchedule(row, leg);
     PaymentSchedule paymentSch = parsePaymentSchedule(row, leg, accrualSch.getFrequency());
     NotionalSchedule notionalSch = parseNotionalSchedule(row, leg);
+
+    Optional<String> knownAmount = findValue(row, leg, KNOWN_AMOUNT_FIELD);
+    if (knownAmount.isPresent()) {
+      return KnownAmountSwapLeg.builder()
+          .payReceive(payReceive)
+          .accrualSchedule(accrualSch)
+          .paymentSchedule(paymentSch)
+          .currency(notionalSch.getCurrency())
```

Comment:
Need to parse currency separately to notional schedule, with notional schedule parsed in `parseRateCalculationLeg`. As it stands, you've made `notional` mandatory for parsing `KnownAmountSwapLeg` which isn't true.